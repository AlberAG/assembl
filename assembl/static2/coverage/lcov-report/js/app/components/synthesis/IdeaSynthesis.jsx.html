<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/synthesis/IdeaSynthesis.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">js/app/components/synthesis</a> IdeaSynthesis.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.74% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/23</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
&nbsp;
import * as React from 'react';
import { Translate } from 'react-redux-i18n';
import { Link } from 'react-router';
&nbsp;
import StatisticsDoughnut from '../debate/common/statisticsDoughnut';
import PostsAndContributorsCount from '../common/postsAndContributorsCount';
import { getColumnInfos, getSentimentsCount, createDoughnutElements } from '../debate/common/announcement';
&nbsp;
export type SynthesisIdea = {
  id: string,
  ancestors: Array&lt;string&gt;,
  title: string,
  synthesisTitle: string,
  live: {
    id: string,
    order: number,
    img: {
      externalUrl: string
    },
    numContributors: number,
    numPosts: number,
    messageColumns: Array&lt;IdeaMessageColumnFragment&gt;,
    messageViewOverride: string,
    posts: {
      edges: Array&lt;Object&gt;
    }
  }
};
&nbsp;
const SynthesisBody = <span class="fstat-no" title="function not covered" >({</span> level, hasSiblings, value, stats }) =&gt; (
<span class="cstat-no" title="statement not covered" >  &lt;div className="synthesis-body" style={{ columnCount: !hasSiblings &amp;&amp; level &gt; 2 ? 2 : 'auto' }}&gt;</span>
    &lt;p dangerouslySetInnerHTML={{ __html: value }} /&gt;
    {stats}
  &lt;/div&gt;
);
&nbsp;
const LinkToIdea = <span class="fstat-no" title="function not covered" >({</span> href }) =&gt; (
<span class="cstat-no" title="statement not covered" >  &lt;Link className="idea-link" to={href}&gt;</span>
    {'&gt; '}
    &lt;Translate value="synthesis.seeConversation" /&gt;
  &lt;/Link&gt;
);
&nbsp;
const SynthesisStats = <span class="fstat-no" title="function not covered" >({</span> numContributors, numPosts, ideaLink, posts, isMultiColumns, messageColumns }) =&gt; {
  const sentimentsCount = <span class="cstat-no" title="statement not covered" >getSentimentsCount(posts);</span>
  const columnInfos = <span class="cstat-no" title="statement not covered" >getColumnInfos(messageColumns);</span>
  const doughnutElements = <span class="cstat-no" title="statement not covered" >isMultiColumns ? columnInfos : createDoughnutElements(sentimentsCount);</span>
<span class="cstat-no" title="statement not covered" >  return (</span>
    &lt;div className="synthesis-stats"&gt;
      &lt;StatisticsDoughnut elements={doughnutElements} placement="after" /&gt;
      &lt;PostsAndContributorsCount vertical numContributors={numContributors} numPosts={numPosts} showNumPosts={!isMultiColumns} /&gt;
      &lt;LinkToIdea href={ideaLink} /&gt;
    &lt;/div&gt;
  );
};
&nbsp;
const SynthesisImage = <span class="fstat-no" title="function not covered" >({</span> level, imgUrl, stats }) =&gt; {
<span class="cstat-no" title="statement not covered" >  if (level === 1) {</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;div className="synthesis-image-container" style={imgUrl &amp;&amp; { backgroundImage: `url(${imgUrl})` }}&gt;
        {stats}
      &lt;/div&gt;
    );
  }
<span class="cstat-no" title="statement not covered" >  if (level === 2) {</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;div&gt;
        &lt;div className="image-container"&gt;
          &lt;div className="synthesis-image-container" style={imgUrl &amp;&amp; { backgroundImage: `url(${imgUrl})` }} /&gt;
        &lt;/div&gt;
        {stats}
      &lt;/div&gt;
    );
  }
<span class="cstat-no" title="statement not covered" >  return null;</span>
};
&nbsp;
const IdeaSynthesis = <span class="fstat-no" title="function not covered" >(p</span>rops: { idea: SynthesisIdea, hasSiblings: boolean, level: number, slug: string }) =&gt; {
  const { idea, hasSiblings, level, slug } = <span class="cstat-no" title="statement not covered" >props;</span>
  const { id, img, numContributors, numPosts, posts, messageViewOverride, messageColumns } = <span class="cstat-no" title="statement not covered" >idea.live;</span>
  const phaseIdentifier = <span class="cstat-no" title="statement not covered" >'thread';</span> // TODO: Proper phase identification
  // For now, syntheses can only have ideas from the "thread" phase.
&nbsp;
  const imgUrl = <span class="cstat-no" title="statement not covered" >img &amp;&amp; img.externalUrl;</span>
  const link = <span class="cstat-no" title="statement not covered" >`/${slug}/debate/${phaseIdentifier}/theme/${id}`;</span>
  const stats = (
<span class="cstat-no" title="statement not covered" >    &lt;SynthesisStats</span>
      numContributors={numContributors}
      numPosts={numPosts}
      ideaLink={link}
      posts={posts}
      isMultiColumns={messageViewOverride === 'messageColumns'}
      messageColumns={messageColumns}
    /&gt;
  );
<span class="cstat-no" title="statement not covered" >  return (</span>
    &lt;div className={`${'idea-synthesis idea-synthesis-level-'}${level}`}&gt;
      &lt;SynthesisImage level={level} imgUrl={imgUrl} stats={stats} /&gt;
      &lt;SynthesisBody value={idea.synthesisTitle} hasSiblings={hasSiblings} level={level} stats={level &gt;= 3 ? stats : null} /&gt;
    &lt;/div&gt;
  );
};
&nbsp;
export default IdeaSynthesis;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
