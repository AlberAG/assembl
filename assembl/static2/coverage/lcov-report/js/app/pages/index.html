<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/pages</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> js/app/pages
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">35.58% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>201/565</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.13% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>97/333</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.11% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>52/185</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.59% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>191/522</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="brightMirrorFiction.jsx"><a href="brightMirrorFiction.jsx.html">brightMirrorFiction.jsx</a></td>
	<td data-value="70" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 70%;"></div><div class="cover-empty" style="width:30%;"></div></div></td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="60" class="abs medium">42/60</td>
	<td data-value="69.05" class="pct medium">69.05%</td>
	<td data-value="42" class="abs medium">29/42</td>
	<td data-value="37.5" class="pct low">37.5%</td>
	<td data-value="16" class="abs low">6/16</td>
	<td data-value="70.91" class="pct medium">70.91%</td>
	<td data-value="55" class="abs medium">39/55</td>
	</tr>

<tr>
	<td class="file high" data-value="community.jsx"><a href="community.jsx.html">community.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file low" data-value="idea.jsx"><a href="idea.jsx.html">idea.jsx</a></td>
	<td data-value="36.92" class="pic low"><div class="chart"><div class="cover-fill" style="width: 36%;"></div><div class="cover-empty" style="width:64%;"></div></div></td>
	<td data-value="36.92" class="pct low">36.92%</td>
	<td data-value="130" class="abs low">48/130</td>
	<td data-value="31.51" class="pct low">31.51%</td>
	<td data-value="73" class="abs low">23/73</td>
	<td data-value="37.93" class="pct low">37.93%</td>
	<td data-value="29" class="abs low">11/29</td>
	<td data-value="38.1" class="pct low">38.1%</td>
	<td data-value="126" class="abs low">48/126</td>
	</tr>

<tr>
	<td class="file medium" data-value="question.jsx"><a href="question.jsx.html">question.jsx</a></td>
	<td data-value="61.54" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 61%;"></div><div class="cover-empty" style="width:39%;"></div></div></td>
	<td data-value="61.54" class="pct medium">61.54%</td>
	<td data-value="13" class="abs medium">8/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="12" class="abs medium">8/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="requestPasswordChange.jsx"><a href="requestPasswordChange.jsx.html">requestPasswordChange.jsx</a></td>
	<td data-value="57.14" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 57%;"></div><div class="cover-empty" style="width:43%;"></div></div></td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	</tr>

<tr>
	<td class="file medium" data-value="syntheses.jsx"><a href="syntheses.jsx.html">syntheses.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="12" class="abs medium">6/12</td>
	<td data-value="37.5" class="pct low">37.5%</td>
	<td data-value="16" class="abs low">6/16</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="54.55" class="pct medium">54.55%</td>
	<td data-value="11" class="abs medium">6/11</td>
	</tr>

<tr>
	<td class="file low" data-value="synthesis.jsx"><a href="synthesis.jsx.html">synthesis.jsx</a></td>
	<td data-value="27.66" class="pic low"><div class="chart"><div class="cover-fill" style="width: 27%;"></div><div class="cover-empty" style="width:73%;"></div></div></td>
	<td data-value="27.66" class="pct low">27.66%</td>
	<td data-value="47" class="abs low">13/47</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="33" class="abs low">3/33</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	<td data-value="28.26" class="pct low">28.26%</td>
	<td data-value="46" class="abs low">13/46</td>
	</tr>

<tr>
	<td class="file medium" data-value="voteSession.jsx"><a href="voteSession.jsx.html">voteSession.jsx</a></td>
	<td data-value="51.33" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 51%;"></div><div class="cover-empty" style="width:49%;"></div></div></td>
	<td data-value="51.33" class="pct medium">51.33%</td>
	<td data-value="113" class="abs medium">58/113</td>
	<td data-value="36" class="pct low">36%</td>
	<td data-value="75" class="abs low">27/75</td>
	<td data-value="44" class="pct low">44%</td>
	<td data-value="50" class="abs low">22/50</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="106" class="abs medium">53/106</td>
	</tr>

<tr>
	<td class="file low" data-value="voteSessionAdmin.jsx"><a href="voteSessionAdmin.jsx.html">voteSessionAdmin.jsx</a></td>
	<td data-value="11.54" class="pic low"><div class="chart"><div class="cover-fill" style="width: 11%;"></div><div class="cover-empty" style="width:89%;"></div></div></td>
	<td data-value="11.54" class="pct low">11.54%</td>
	<td data-value="182" class="abs low">21/182</td>
	<td data-value="8.14" class="pct low">8.14%</td>
	<td data-value="86" class="abs low">7/86</td>
	<td data-value="4.76" class="pct low">4.76%</td>
	<td data-value="63" class="abs low">3/63</td>
	<td data-value="11.88" class="pct low">11.88%</td>
	<td data-value="160" class="abs low">19/160</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
