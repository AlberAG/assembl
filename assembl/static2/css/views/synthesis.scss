.synthesis-page {
  @media screen and (max-width: 1000px) {
    .synthesis-side-menu {
      display: none;
    }
  }

  .synthesis-side-menu {
    background-color: $white;
    padding: 30px 10px 30px 30px;

    .title-hyphen {
      display: block;
      width: 50px;
      margin: 10px 0 10px -35px;
    }

    .active-caret {
      transform: rotate(180deg);
    }

    .link-container {
      display: flex;
      align-items: center;
      justify-content: space-between;
    }

    .active {
      background: rgba($first-color, 0.15);

      a {
        pointer-events: none;
      }
    }

    .side-menu-link-1 {
      overflow: auto;
      font-size: $font-size-s;
      color: $first-color;
      line-height: 22px;
      font-family: $font-family3;
      letter-spacing: 2px;
      padding: 4px 0;
    }

    .side-menu-link-2 {
      overflow: auto;
      margin-left: 20px;
      font-size: $font-size-xs;
      line-height: 20px;
      font-family: $font-family3;
    }

    .side-menu-link-3 {
      overflow: auto;
      margin-left: 40px;
      font-size: $font-size-xs;
      line-height: 20px;
      font-family: $font-family3;
    }

    .dark-title-4 {
      font-family: $font-family;
      font-size: $font-size-m;
    }
  }

  .header-section {
    height: 500px;
  }

  .synthesis-block .content-section {
    padding: 0 0 60px 0;
    text-align: justify;
    font-size: $font-size-m;
  }

  .synthesis-tree {
    padding-top: 60px;
  }

  .idea-synthesis-section {
    .content-section {
      padding: 0;
    }

    .title-section {
      margin: 0;
    }

    .section-title1 {
      font-size: 32px;
    }

    .section-title-2 {
      font-size: 26px;
    }

    .section-title-3 {
      font-size: 20px;
    }
  }
}
