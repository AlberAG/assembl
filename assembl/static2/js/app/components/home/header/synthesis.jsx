import React from 'react';
import { connect } from 'react-redux';
import { Translate, Localize } from 'react-redux-i18n';
import { getConnectedUserId, getDiscussionSlug } from '../../../utils/globalFunctions';
import { getCurrentView, get, getContextual } from '../../../utils/routeMap';

class Synthesis extends React.Component {
  render() {
    const { synthesis } = this.props.synthesis;
    const slug = { slug: getDiscussionSlug() };
    const connectedUserId = getConnectedUserId();
    const next = getCurrentView();
    return (
      <div className="synthesis-container">
        {synthesis &&
          Object.keys(synthesis.lastPublishedSynthesis).length > 0 &&
          synthesis.lastPublishedSynthesis.introduction &&
          <a
            href={
              connectedUserId
                ? `${get('oldDebate', slug)}/posts/${synthesis.lastPublishedSynthesis.published_in_post}`
                : `${getContextual('login', slug)}?next=${next}`
            }
          >
            <div className="insert-box">
              <h3 className="dark-title-4 ellipsis">
                <div>
                  <Translate value="synthesis.title" />
                </div>
                <div className="ellipsis margin-xs">
                  {synthesis.lastPublishedSynthesis.subject}
                </div>
              </h3>
              <div className="box-hyphen">&nbsp;</div>
              <div className="date">
                <Localize value={synthesis.lastPublishedSynthesis.creation_date} dateFormat="date.format2" />
              </div>
              <div className="insert-content margin-s">
                {<p dangerouslySetInnerHTML={{ __html: synthesis.lastPublishedSynthesis.introduction }} />}
              </div>
            </div>
          </a>}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    synthesis: state.synthesis
  };
};

export default connect(mapStateToProps)(Synthesis);