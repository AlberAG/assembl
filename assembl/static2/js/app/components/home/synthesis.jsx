import React from 'react';
import { connect } from 'react-redux';
import { Grid, Row, Col } from 'react-bootstrap';
import MapStateToProps from '../../store/mapStateToProps';
import Synthesis from './header/synthesis';

class SynthesisContainer extends React.Component {
  render() {
    const { synthesis } = this.props.synthesis;
    return (
      <section className="synthesis-section">
        <Grid fluid className="background-light">
          <div className="max-container">
            <div style={{ margin: '20px 0' }}>
              <Row>
                <Col md={12}>
                  {synthesis && Object.keys(synthesis.lastPublishedSynthesis).length > 0 &&
                    <Synthesis />
                  }
                </Col>
              </Row>
            </div>
          </div>
        </Grid>
      </section>
    );
  }
}

export default connect(MapStateToProps)(SynthesisContainer);