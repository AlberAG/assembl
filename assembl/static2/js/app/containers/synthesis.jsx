import React from 'react';
import { Translate } from 'react-redux-i18n';

const Synthesis = () => (
  <p><Translate value="synthesis.panelTitle" /></p>
);

export default Synthesis;