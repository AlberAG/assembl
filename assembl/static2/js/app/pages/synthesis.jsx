// @flow
import React from 'react';
import { Translate } from 'react-redux-i18n';
import { connect } from 'react-redux';
import { compose, graphql } from 'react-apollo';

import SynthesisQuery from '../graphql/SynthesisQuery.graphql';
import withLoadingIndicator from '../components/common/withLoadingIndicator';
import IdeaSynthesis from '../components/synthesis/IdeaSynthesis';

type SynthesisProps = {
  synthesis: Object
};

export class DumbSynthesis extends React.Component<void, SynthesisProps, void> {
  props: SynthesisProps;
  render() {
    const { synthesis } = this.props;
    return (
      <div className="max-container">
        <Translate value="synthesis.title" />
        <IdeaSynthesis {...synthesis} />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    lang: state.i18n.locale
  };
};

export default compose(
  connect(mapStateToProps),
  graphql(SynthesisQuery, {
    options: (props) => {
      return {
        variables: { id: props.params.synthesisId }
      };
    },
    props: ({ data }) => {
      if (data.loading) {
        return { dataLoading: true };
      }
      return {
        synthesis: data.synthesis
      };
    }
  }),
  withLoadingIndicator()
)(DumbSynthesis);