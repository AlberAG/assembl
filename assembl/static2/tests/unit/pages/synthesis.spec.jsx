import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { DumbSynthesis } from '../../../js/app/pages/synthesis';

describe('Synthesis component', () => {
  it('should match Synthesis snapshot', () => {
    const props = {
      routeParams: { slug: 'slugId' },
      synthesis: {
        id: 'fooId',
        subject: 'Foo',
        imgUrl: 'http://foo.com/foo',
        ideas: [
          {
            id: 'barId',
            title: 'Bar',
            imgUrl: 'http://foo.com/bar',
            synthesisTitle: 'Foo',
            numContributors: 0,
            numPosts: 0,
            posts: { edges: [] },
            phaseIdentifier: 'paseId'
          }
        ]
      }
    };
    const renderer = new ShallowRenderer();
    renderer.render(<DumbSynthesis {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});