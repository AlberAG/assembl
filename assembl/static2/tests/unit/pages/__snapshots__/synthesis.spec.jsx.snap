// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Synthesis component should match Synthesis snapshot 1`] = `
<div
  className="max-container"
>
  <Connect(Header)
    imgUrl="http://foo.com/foo"
    isSynthesesHeader={true}
    title="Foo"
  />
  <Unknown
    ancestors={Array []}
    id="barId"
    imgUrl="http://foo.com/bar"
    index={1}
    numContributors={0}
    numPosts={0}
    parents={Array []}
    phaseIdentifier="paseId"
    posts={
      Object {
        "edges": Array [],
      }
    }
    slug="slugId"
    subIdeas={Array []}
    synthesisTitle="Foo"
    title="Bar"
  />
</div>
`;
